### PR TITLE
Add --force to test scaffold

### LINKED ISSUE #31455: Add --force option to test scaffold
This is a follow up issue to #31260.

This parameter will all interactive bypass of confirmation prompts and allow for unattended use. Usage should be 

`pnpx wc-e2e install @woocommerce/e2e-core-tests --force`

### PR DESCRIPTION
**Pull request by Luc45**
This PR adds an optional `--force` parameter to the test scaffold that allows to run the test scaffold, overriding local files without prompting a confirmation to the user if the files already exist.

Closes #31455

### How to test:

1. Run `cd plugins/woocommerce`
2. Run `pnpm install`
3. Rename the tests folder: `mv tests tests-x`
4. Run `pnpx wc-e2e install`
5. This should create
```
tests/e2e/config/jest.config.js
tests/e2e/config/jest.setup.js
tests/e2e/docker/initialize.sh
```
6. Run `pnpx wc-e2e install --force`
7. This should re-create the files above without prompting a confirmation

### Testing that the test files are created without a prompt:

8. As a continuation to the tests above, run `pnpx wc-e2e install @woocommerce/e2e-core-tests`
9. This should create many test files under the folder `tests/e2e/specs`
10. Now run `pnpx wc-e2e install @woocommerce/e2e-core-tests --force`
10. This should re-create the test files inside the `tests/e2e/specs` folder without prompting a confirmation

### CODE REVIEW

**Comment by rrennick** [comment]
@Luc45 Could you add test instructions to verify this works with installing e2e-core-tests as well?

File: packages/js/e2e-environment/bin/scaffold.js [hunk 1]
```diff
@@ -24,12 +24,24 @@ const {
 } = require( '../utils/scaffold' );
 
 const args = process.argv.slice( 2 );
-const [ command, packageName ] = args;
+const command = args[0];
+let packageName = null;
+
+if (args[1].substr(0, 2) !== '--') {
```
Comment (MrJnrman) [inline packages/js/e2e-environment/bin/scaffold.js]: This check here causes a break when I run the ` pnpx wc-e2e install ` command. Doing a check to ensure that `args[1]` isn't undefined seems to do the trick. 
```suggestion
if ( args[1] && args[1].substr(0, 2) !== '--') {
```

**Review by MrJnrman — CHANGES_REQUESTED** [review]
Overall great job here @Luc45! I was able to follow your instructions and used `--force` parameter to override local files using the test scaffold.

I only have one suggestion which I noted in a comment.

File: packages/js/e2e-environment/bin/scaffold.js (hunk 1 above)
Comment (Luc45) [reply packages/js/e2e-environment/bin/scaffold.js]: Thanks for the suggestion, @MrJnrman! I can see how this is necessary, thank you. If there's no package name, I'm potentially calling `substr` on `undefined`.

I'm just curious to understand why I didn't face this error in my local test when running `pnpx wc-e2e install` :thinking: I'm running Node.js version v16.13.1 for what it's worth

**Comment by Luc45** [comment]
> @Luc45 Could you add test instructions to verify this works with installing e2e-core-tests as well?

Thanks for pointing that out, the test files were added to the --force parameter as well. I have added test instructions to the main thread of this PR. (Steps 8 to 11)

File: packages/js/e2e-environment/bin/scaffold.js [hunk 2]
```diff
@@ -24,12 +24,24 @@ const {
 } = require( '../utils/scaffold' );
 
 const args = process.argv.slice( 2 );
-const [ command, packageName ] = args;
+const command = args[0];
+let packageName = null;
+
+if ( args[1] && args[1].substr(0, 2) !== '--' ) {
```
Comment (MrJnrman) [inline packages/js/e2e-environment/bin/scaffold.js]: Small nitpick with the spacing.

```suggestion
if ( args[ 1 ] && args[ 1 ].substr( 0, 2 ) !== '--' ) {
```
Using a linter can help with this

File: packages/js/e2e-environment/bin/scaffold.js [hunk 3]
```diff
@@ -24,12 +24,24 @@ const {
...
+	packageName = args[1];
+}
 
 // Allow multiple spec file extensions and formats.
 let testExtension = 'test.js';
 let testFormat = '';
-for ( let a = 2; a < args.length; a++ ) {
+let force = false;
+for ( let a = 1; a < args.length; a++ ) {
+	if (args[a] === '--force') {
```
Comment (MrJnrman) [inline packages/js/e2e-environment/bin/scaffold.js]: Adjust spacing

File: packages/js/e2e-environment/bin/scaffold.js [hunk 4]
```diff
@@ -50,7 +62,7 @@ for ( let a = 2; a < args.length; a++ ) {
 if ( command == 'install' ) {
 	// Install some environment defaults if no package is requested.
 	if ( ! packageName ) {
-		installDefaults();
+		installDefaults(force);
```
Comment (MrJnrman) [inline packages/js/e2e-environment/bin/scaffold.js]: Spacing

File: packages/js/e2e-environment/bin/scaffold.js [hunk 5]
```diff
@@ -95,13 +107,16 @@ if ( command == 'install' ) {
...
 		let importLineFormat;
-		let overwriteFiles;
 		let confirmPrompt;
 		if ( testFormat.toLowerCase() == 'cjs' ) {
 			importLineFormat = sprintf( "const {%%s} = require( '%s' );", pkg );
 		} else {
 			importLineFormat = sprintf( "import {%%s} from '%s';", pkg );
 		}
+		let overwriteFiles;
+		if (force) {
```
Comment (MrJnrman) [inline packages/js/e2e-environment/bin/scaffold.js]: Spacing

File: packages/js/e2e-environment/utils/scaffold.js [hunk 6]
```diff
@@ -102,16 +103,16 @@ const getPackageData = ( packageName ) => {
 /**
  * Install test runner and test container defaults
  */
-const installDefaults = () => {
+const installDefaults = (force) => {
```
Comment (MrJnrman) [inline packages/js/e2e-environment/utils/scaffold.js]: Spacing

**Review by MrJnrman — CHANGES_REQUESTED** [review]
Thanks for making the changes @Luc45 . I was able to run `pnpx wc-e2e install` & `pnpx wc-e2e install @woocommerce/e2e-core-tests` with/without `--force` successfully. And the test suite was able to be executed without a hitch. 

I just have a few small nitpicks with the spacing and this can be merged.

**Review by MrJnrman — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**